proto/event.proto

```diff
@@ -42,6 +48,13 @@ message Counter {
   map<string, string> tags = 4;
 }
 
+message AggregatedCounter {
+  string name = 1;
+  double val = 2;
+  google.protobuf.Timestamp timestamp = 3;
+  map<string, string> tags = 4;
+}
```

Comment:
Do you think we need this in addition to the normal counter? They seem equivalent.

Reply:
Yeah, that's something I'd like to discuss. Let me give you an example.

Imagine we have a metric that is incremented every second. And we have two interfaces for consuming this metric into Vector. First is a Statsd socket, and second is a Prometheus exposition. When we connect those two as sinks, we will start observing the data like this:

| Time (s) | Statsd Counter | Prometheus Counter |
|----------|----------------|--------------------|
| 5        | 5              | 5                  |
| 10       | 5              | 10                 |
| 15       | 5              | 15                 |
| ..       | ..             | ..                 |

The idea here was to call Statsd counter just `Counter`, and a Prometheus one `AggregatedCounter`, to make sure we do not do double aggregation of metrics.

I imagine also we'll need to be able to convert between two kinds.

For instance, what our Prometheus sink is currently doing is aggregation of counters.

And in order to route e.g. Prometheus counters into Statsd sink we will need to disaggregate them somehow, calculate those delta values (or convert into Gauge).

Reply:
I understand it's too much for a single PR, I'm sorry! I just don't really know how to possibly break this down, there are so many edge cases I'd like to explore before pushing the changes.

Reply:
I see, that makes sense! I'm wondering if it makes sense to split incremental vs aggregated metrics apart one level higher, and then maybe we could reuse internal representations?

We could also add support for telling the different into the type system between components so that we could disallow things like statsd from consuming prometheus without first doing some conversion to incremental.

Not sure about any of these, just brainstorming.